File: ExcAlgoritmoProgramacao_JS/Lista_Bee/1478_basico.js
import { question } from "readline-sync";

function main() {
    let n = Number(question());

    while(n !== 0) {
        let matriz = [];

        if(n >= 0 && n <= 100) {
            for (let i = 0; i < n; i++) {
                let linha = [];
    
                for (let j = 0; j < n; j++) {
                    let valor = Math.abs(i - j) + 1;
                    linha.push(valor);
        
                }
                matriz.push(linha);
        
            }
            for (let i = 0; i < matriz.length; i++) {
                let linha = '';
                for (let j = 0; j < matriz[i].length; j++) {
                    linha += matriz[i][j].toString().padStart(3, ' ') + (j === matriz[i].length - 1 ? '' : ' ');
                }
                console.log(linha);
            }
            console.log('');
            matriz = [];
            
        }
        n = Number(question());

    }
}

main();

File: ExcAlgoritmoProgramacao_JS/Lista_Bee/1478_basico.py
def main ():
    n = get_Number()

    while (n != 0):
        matriz = []

        for i in range(n):
            linha = []

            for j in range(n):
                valor = abs(i - j) + 1
                linha.append(valor)
            
            matriz.append(linha)
        
        for k in range(len(matriz)):
            linhas = ''

            for m in range(len(matriz[k])):
                linhas += str(matriz[k][m]).rjust(3, ' ') + (' ' if m < len(matriz[k]) - 1 else '')

            print(linhas)
        
        print()
        n = get_Number()

def get_Number():
    return int(input().strip())

main()